dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/CopyActionlet.java

```diff
@@ -0,0 +1,129 @@
+import com.dotmarketing.portlets.workflows.model.WorkflowActionletParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+import com.dotmarketing.util.Logger;
+import com.liferay.portal.model.User;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Copy the edited contentlet
+ * @author jsanca
+ */
+public class CopyActionlet extends WorkFlowActionlet {
+
+    private static final long serialVersionUID              = 1L;
+    public static final String NOTIFY_SYNC_COPY_EVENT       = "notify.sync.copy.event";
+    private final HostAPI       hostAPI                     = APILocator.getHostAPI();
+    private final FolderAPI     folderAPI                   = APILocator.getFolderAPI();
+    private final ContentletAPI contentletAPI               = APILocator.getContentletAPI();
+    private final LocalSystemEventsAPI localSystemEventsAPI = APILocator.getLocalSystemEventsAPI();
+
+
+
+    @Override
+    public List<WorkflowActionletParameter> getParameters() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public String getName() {
+        return "Copy Contentlet";
+    }
+
+    @Override
+    public String getHowTo() {
+        return "This workflow actionlet copies the edited contentlet ";
+    }
+
+    @Override
+    public void executePreAction(final WorkflowProcessor processor,
+                    final Map<String, WorkflowActionClassParameter> params)
+                    throws WorkflowActionFailureException, DotContentletValidationException {
+
+        final  Contentlet contentlet = processor.getContentlet();
+
+        try {
+
+            if (!contentlet.isLocked()) {
+
+                this.performCopy(contentlet, processor.getUser());
+            }
+        } catch (DotContentletStateException | DotDataException | DotSecurityException | IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new WorkflowActionFailureException(e.getMessage(), e);
+        }
+    } // executePreAction.
+
+    @Override
+    public void executeAction(final WorkflowProcessor processor,
+                    final Map<String, WorkflowActionClassParameter> params)
+                    throws WorkflowActionFailureException {
+
+        final Contentlet contentlet = processor.getContentlet();
+
+        try {
+
+            if (contentlet.isLocked()) {
+
+                this.performCopy(contentlet, processor.getUser());
+            }
+        } catch (DotContentletStateException | DotDataException | DotSecurityException | IOException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new WorkflowActionFailureException(e.getMessage(), e);
+        }
+    } // executeAction.
+
+    private void performCopy(final Contentlet contentlet,
+                             final User user) throws DotDataException, DotSecurityException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)